File: home/combinador.java
import java.util.Scanner;

public class combinador {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        int N = scanner.nextInt();
        scanner.nextLine(); // Consumir a nova linha após o inteiro

        for (int i = 0; i < N; i++) {
            String str1 = scanner.nextLine();
            String str2 = scanner.nextLine();
            System.out.println(combinarStrings(str1, str2));
        }

        scanner.close();
    }

    private static String combinarStrings(String str1, String str2) {
        StringBuilder resultado = new StringBuilder();
        int len1 = str1.length();
        int len2 = str2.length();
        int minLen = Math.min(len1, len2);
        
        // Alternando letras
        for (int i = 0; i < minLen; i++) {
            resultado.append(str1.charAt(i));
            resultado.append(str2.charAt(i));
        }
        
        // Adicionando letras restantes da string mais longa
        if (len1 > minLen) {
            resultado.append(str1.substring(minLen));
        } else {
            resultado.append(str2.substring(minLen));
        }
        
        return resultado.toString();
    }
}

File: home/combinador.c
#include <stdio.h>
#include <string.h>

#define MAX_LENGTH 51 // Máximo de 50 caracteres + 1 para o terminador nulo

void combinarStrings(const char* str1, const char* str2, char* resultado) {
    int len1 = strlen(str1);
    int len2 = strlen(str2);
    int j = 0;

    // Alterna as letras das duas strings
    for (int i = 0; i < len1 || i < len2; i++) {
        if (i < len1) {
            resultado[j++] = str1[i]; // Adiciona letra da primeira string
        }
        if (i < len2) {
            resultado[j++] = str2[i]; // Adiciona letra da segunda string
        }
    }

    resultado[j] = '\0'; // Termina a string
}

int main() {
    int N;
    
    // Lê o número de casos de teste
    if (scanf("%d\n", &N) != 1 || N < 1) {
        fprintf(stderr, "Número de casos de teste inválido.\n");
        return 1;
    }

    for (int t = 0; t < N; t++) {
        char str1[MAX_LENGTH], str2[MAX_LENGTH], resultado[MAX_LENGTH * 2]; // Resultado pode ter até 100 caracteres

        // Lê as duas strings
        if (fgets(str1, sizeof(str1), stdin) == NULL || fgets(str2, sizeof(str2), stdin) == NULL) {
            fprintf(stderr, "Erro ao ler as strings.\n");
            return 1;
        }

        // Remove o caractere de nova linha, se presente
        str1[strcspn(str1, "\n")] = 0;
        str2[strcspn(str2, "\n")] = 0;

        // Combina as strings
        combinarStrings(str1, str2, resultado);
        
        // Imprime a string resultante
        printf("%s\n", resultado);
    }

    return 0;
}
